core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-add-variable/store/dot-add-variable.store.ts

```diff
@@ -7,48 +7,126 @@ import { Injectable } from '@angular/core';
 import { catchError, switchMap, tap } from 'rxjs/operators';
 
 import { DotGlobalMessageService } from '@components/_common/dot-global-message/dot-global-message.service';
-import { DotContentTypeService } from '@dotcms/data-access';
+import { DotContentTypeService, DotMessageService } from '@dotcms/data-access';
 import { DotCMSContentType, DotCMSContentTypeField } from '@dotcms/dotcms-models';
 import { DotHttpErrorManagerService } from '@services/dot-http-error-manager/dot-http-error-manager.service';
 
+import { DotFieldContent, FilteredFieldTypes } from '../dot-add-variable.models';
+
 export interface DotAddVariableState {
-    variables: DotCMSContentTypeField[];
+    fields: DotFieldContent[];
 }
 
 @Injectable()
 export class DotAddVariableStore extends ComponentStore<DotAddVariableState> {
     constructor(
         private dotContentTypeService: DotContentTypeService,
         private dotGlobalMessageService: DotGlobalMessageService,
-        private dotHttpErrorManagerService: DotHttpErrorManagerService
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private dotMessage: DotMessageService
     ) {
         super({
-            variables: []
+            fields: []
         });
     }
 
-    readonly vm$ = this.select(({ variables }) => {
+    readonly vm$ = this.select(({ fields }) => {
         return {
-            variables
+            fields
         };
     });
 
-    readonly updateVariables = this.updater<DotCMSContentTypeField[]>(
-        (state: DotAddVariableState, variables: DotCMSContentTypeField[]) => {
+    // You can add here a new fieldType and add the fields that it has
+    private readonly fields: Record<
+        string,
+        (variableContent: DotFieldContent) => DotFieldContent[]
+    > = {
+        Image: ({ variable, name, fieldTypeLabel }) => [
```

Comment:
What about binary, file, maybe others did you check?

Reply:
Maybe we should move this out of the store and import it... so the store don't get huge.

Reply:
Made a service for this and I will make research for all the fields that have custom code and I'll add them